### PR TITLE
Sensor Card: Use Entity card with graph footer

### PR DESCRIPTION
**Pull request by zsarnett**
## Proposed change

Changes the sensor card to be an entity card and if a Line graph is requested. Add a graph footer to the entity card

It depends on the  #4971 PR to be merged.

Only file that was actually changed in this PR is the Sensor Card file.

## Type of change

- [x] Code quality improvements to existing code or addition of tests

## Additional information

The same configuration can be used. Nothing will break. Sensor card will continue to be an option

- This PR is related to the issue: #4971 

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [x] Tests have been added to verify that the new code works.

### CODE REVIEW

**Comment by zsarnett** [comment]
@bramkragten if there isn't too much discussion on this I think it would be good for 0.108

File: src/panels/lovelace/cards/hui-sensor-card.ts [hunk 1]
```diff
@@ -202,304 +43,35 @@ class HuiSensorCard extends LitElement implements LovelaceCard {
...
-      theme: "default",
-      hours_to_show: 24,
-      ...config,
+    let entityCardConfig: EntityCardConfig = {
+      type: "entity",
+      entity: config.entity,
+      theme: config.theme || "default",
+      ...(config.name && { name: config.name }),
+      ...(config.unit && { unit: config.unit }),
+      ...(config.icon && { icon: config.icon }),
```
Comment (bramkragten) [inline src/panels/lovelace/cards/hui-sensor-card.ts]: Can't we do:
```suggestion
    let entityCardConfig: EntityCardConfig = {
      ...config,
      type: "entity",
```

File: src/panels/lovelace/cards/hui-sensor-card.ts [hunk 2]
```diff
@@ -202,304 +43,35 @@ class HuiSensorCard extends LitElement implements LovelaceCard {
...
+        type: "graph",
+        entity: config.entity,
+        detail: config.detail || 1,
+        hours_to_show: config.hours_to_show || 24,
+      };
+
+      entityCardConfig = {
+        ...entityCardConfig,
+        footer: footerConfig,
+      };
```
Comment (bramkragten) [inline src/panels/lovelace/cards/hui-sensor-card.ts]: No need to make a copy: `entityCardConfig.footer = footerConfig`

File: src/panels/lovelace/cards/hui-sensor-card.ts [hunk 3]
```diff
@@ -202,304 +43,35 @@ class HuiSensorCard extends LitElement implements LovelaceCard {
...
 
-    const cardConfig = {
-      detail: 1,
-      theme: "default",
-      hours_to_show: 24,
-      ...config,
+    let entityCardConfig: EntityCardConfig = {
+      type: "entity",
+      entity: config.entity,
+      theme: config.theme || "default",
```
Comment (bramkragten) [inline src/panels/lovelace/cards/hui-sensor-card.ts]: We shouldn't add `theme: "default"` when no theme is set

File: src/panels/lovelace/cards/hui-sensor-card.ts (hunk 1 above)
Comment (zsarnett) [reply src/panels/lovelace/cards/hui-sensor-card.ts]: We can sure. But then we are sending in extra config variables

Graph, detail, hours to show

Comment (bramkragten) [reply src/panels/lovelace/cards/hui-sensor-card.ts]: So pull those out for the footer first and then send the rest to the `entityCard`

**Review by bramkragten — APPROVED** [review]

**Pull request merged**